### PR TITLE
Feat: authentiation(Register and Login)

### PR DESCRIPTION
**Pull request by Rohitprasad83**
### Authentication 
- Signup using FirstName, LastName, Email, Password
- Login using email and password
- Signup using dummy details
- Login using dummy details

## Ignore
- index.js file:- will put all the providers in one file later on.

### CODE REVIEW

File: src/pages/authentication/Login.jsx [hunk 1]
```diff
@@ -1,30 +1,66 @@
...
+      const response = await axios.post('/api/auth/login', {
+        email,
+        password,
+      })
+      localStorage.setItem('token', response.data.encodedToken)
+      setUsers(response.data.foundUser)
+      response.status === 200 && navigation('/')
+    } catch (err) {
+      setError("Could'nt Login Up, Please try Again!")
+      console.log(err)
```
Comment (Aditi0505) [inline src/pages/authentication/Login.jsx:27]: Remove stray console.log

File: src/pages/authentication/Register.jsx [hunk 2]
```diff
@@ -1,79 +1,200 @@
...
+        firstName,
+        lastName,
+        password,
+      })
+      localStorage.setItem('token', response.data.encodedToken)
+      setUsers(response.data.createdUser)
+      response.status === 201 && navigation('/')
+    } catch (err) {
+      setError("Could'nt Sign Up, Please try Again!")
+      console.log(err)
```
Comment (Aditi0505) [inline src/pages/authentication/Register.jsx:39]: Same as above.

**Review by Aditi0505 — APPROVED** [review]
- Clean and easily readable code.
- Good folder structure.
- Proper validation of email and password fields.

File: src/pages/authentication/Login.jsx (hunk 1 above)
Comment (Rohitprasad83) [reply src/pages/authentication/Login.jsx:27]: will remove it once i implement Toast

File: src/pages/authentication/Register.jsx (hunk 2 above)
Comment (Rohitprasad83) [reply src/pages/authentication/Register.jsx:39]: Sure will do it

**Pull request merged**